arch/risc-v/src/fe310/fe310_irq_dispatch.c

```diff
@@ -72,7 +72,7 @@ void *fe310_dispatch_irq(uint32_t vector, uint32_t *regs)
 
   if (FE310_IRQ_ECALLM == irq)
     {
-      *mepc += 2;
+      *mepc += 4;
```

Comment:
Why this is changed?

Reply:
I find that ecall instruction is 4 byte even with C (compress) extension now by disassembly, so we must update the epc adjustment.

Reply:
I don't know if this was ever needed for RV32 chips since the ECALL handler does the same thing: 
https://github.com/apache/incubator-nuttx/blob/1fd51ccbe244b554386baa78d1922af5dd89d835/arch/risc-v/src/rv32im/riscv_swint.c#L138-L141

For instance, we didn't do that for ESP32-C3, but we couldn't change it for other chips since we don't have boards to test.

Reply:
We can test it with FE310 and qemu-rv32 on QEMU.

Reply:
Do you have the test result?